gnocchi/rest/aggregates/api.py

```diff
@@ -0,0 +1,209 @@
+    return start, stop, granularity, needed_overlap, fill
+
+
+def _OperationsSubNodeSchema(v):
+    return OperationsSubNodeSchema(v)
+
+
+def MetricSchema(v):
+    """metric keyword schema
+
+    It could be:
+
+    ["metric", "metric-ref", "aggregation"]
+
+    or
+
+    ["metric, ["metric-ref", "aggregation"], ["metric-ref", "aggregation"]]
+    """
+    if not isinstance(v, (list, tuple)) or len(v) <= 2 or v[0] != u"metric":
+        raise voluptuous.Invalid("'metric' is invalid")
+    return [u"metric"] + voluptuous.Schema(voluptuous.Any(
+        voluptuous.ExactSequence([six.text_type, six.text_type]),
+        voluptuous.All(
+            voluptuous.Length(min=1),
+            [voluptuous.ExactSequence([six.text_type, six.text_type])],
+        )), required=True)(v[1:])
+
+
+OperationsSchemaBase = [
+    MetricSchema,
+    voluptuous.ExactSequence(
+        [u"aggregate",
+         voluptuous.Any(*list(agg_operations.AGG_MAP.keys())),
+         _OperationsSubNodeSchema]
+    ),
+]
+
+
+OperationsSubNodeSchema = voluptuous.Schema(voluptuous.Any(*tuple(
+    OperationsSchemaBase + [voluptuous.Coerce(float)]
+)), required=True)
+
+
+def OperationsSchema(v):
+    if isinstance(v, six.text_type):
+        try:
+            v = pyparsing.OneOrMore(
+                pyparsing.nestedExpr()).parseString(v).asList()[0]
+        except pyparsing.ParseException as e:
+            raise ValueError("Invalid value for operations: %s" %
+                             six.text_type(e))
+    return voluptuous.Schema(voluptuous.Any(*OperationsSchemaBase),
+                             required=True)(v)
+
+
+def extract_references(nodes):
+    references = set()
+    if nodes[0] == "metric":
+        if isinstance(nodes[1], list):
+            for subnodes in nodes[1:]:
+                references.add(tuple(subnodes))
+        else:
+            references.add(tuple(nodes[1:]))
+    else:
+        for subnodes in nodes[1:]:
+            if isinstance(subnodes, list):
+                references |= extract_references(subnodes)
+    return references
+
+
+def get_measures_or_abort(metrics_and_aggregations, operations, start,
+                          stop, granularity, needed_overlap, fill,
+                          ref_identifier):
+    try:
+        return processor.get_measures(
+            pecan.request.storage,
+            metrics_and_aggregations,
+            operations,
+            start, stop,
+            granularity, needed_overlap, fill,
+            ref_identifier=ref_identifier)
+    except processor.UnAggregableTimeseries as e:
+        api.abort(400, e)
+    # TODO(sileht): We currently got only one metric for these exceptions but
+    # we can improve processor to returns all missing metrics at once, so we
+    # returns a list for the future
+    except storage.MetricDoesNotExist as e:
+        api.abort(404, {"cause": "Unknown metrics",
+                        "detail": [str(e.metric.id)]})
```

Comment:
why not adding `jsonify` to the class?

Reply:
It's a list of metrics, not just one metric